File: rename_types.py
import os
import re

replacements = {
    'OpenRouterMessage': 'StandardMessage',
    'OpenRouterToolCall': 'StandardToolCall',
    'OpenRouterTool': 'StandardTool',
    'OpenRouterStreamChunk': 'StandardStreamChunk',
    'OpenRouterResponse': 'StandardResponse'
}

def process_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()
        
    original = content
    
    # 1. Rename types
    for old, new in replacements.items():
        content = content.replace(old, new)
        
    # 2. Fix imports from ../api/openrouter
    # We want to change imports of Standard* from openrouter to base-client
    # Let's find imports like: import { StandardMessage, OpenRouterClient } from "../api/openrouter.js"
    # It's easier to just do it via regex substitution
    
    def replacer(match):
        imports = match.group(1)
        path = match.group(2)
        
        has_client = 'OpenRouterClient' in imports
        
        if has_client:
            other_imports = re.sub(r'OpenRouterClient,?\s*', '', imports).strip()
            new_path = path.replace('openrouter', 'base-client')
            
            res = f'import {{ OpenRouterClient }} from "{path}";\n'
            if other_imports:
                # Remove trailing comma if any
                if other_imports.endswith(','): other_imports = other_imports[:-1]
                res += f'import {{ {other_imports} }} from "{new_path}";'
            return res
        else:
            new_path = path.replace('openrouter', 'base-client')
            return match.group(0).replace(path, new_path)

    content = re.sub(r'import\s+\{([^}]+)\}\s+from\s+[\'"](.*api/openrouter.*)[\'"]', replacer, content)
    content = re.sub(r'import\s+\{([^}]+)\}\s+from\s+[\'"](\./openrouter.*)[\'"]', replacer, content)

    if content != original:
        with open(filepath, 'w') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, _, files in os.walk('./src'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            process_file(os.path.join(root, file))

File: rename_types.js
const fs = require('fs');
const path = require('path');
const glob = require('tiny-glob'); // No wait, Tiny glob might not be installed globally, I'll just use a recursive function.

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

const replacements = [
  ['OpenRouterMessage', 'StandardMessage'],
  ['OpenRouterToolCall', 'StandardToolCall'],
  ['OpenRouterTool', 'StandardTool'],
  ['OpenRouterStreamChunk', 'StandardStreamChunk'],
  ['OpenRouterResponse', 'StandardResponse'],
];

walkDir('./src', (filePath) => {
  if (filePath.endsWith('.ts') || filePath.endsWith('.tsx')) {
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;
    
    // Replace names
    for (let [oldName, newName] of replacements) {
      content = content.split(oldName).join(newName);
    }
    
    // Replace imports from openrouter to base-client
    content = content.replace(/import\s+\{([^}]+)\}\s+from\s+['"](?:\.\.\/)+api\/openrouter(?:\.js)?['"]/g, (match, imports) => {
      // If the import contains OpenRouterClient, we have to split it, else just rename the path
      if (imports.includes('OpenRouterClient')) {
        const otherImports = imports.replace(/OpenRouterClient,?\s*/, '').trim();
        let newImport = `import { OpenRouterClient } from "${match.match(/['"](.*?)['"]/)[1]}";\n`;
        if (otherImports) {
            // Need to calculate the relative path to base-client based on the openrouter path
            const openrouterPath = match.match(/['"](.*?)['"]/)[1];
            const baseClientPath = openrouterPath.replace('openrouter', 'base-client');
            newImport += `import { ${otherImports} } from "${baseClientPath}";`;
        }
        return newImport;
      } else {
        const openrouterPath = match.match(/['"](.*?)['"]/)[1];
        const baseClientPath = openrouterPath.replace('openrouter', 'base-client');
        return match.replace(openrouterPath, baseClientPath);
      }
    });

	// Similar for single directory relative imports
	content = content.replace(/import\s+\{([^}]+)\}\s+from\s+['"]\.\/openrouter(?:\.js)?['"]/g, (match, imports) => {
      if (imports.includes('OpenRouterClient')) {
        const otherImports = imports.replace(/OpenRouterClient,?\s*/, '').trim();
        let newImport = `import { OpenRouterClient } from "./openrouter.js";\n`;
        if (otherImports) {
            newImport += `import { ${otherImports} } from "./base-client.js";`;
        }
        return newImport;
      } else {
        return match.replace('./openrouter.js', './base-client.js').replace('./openrouter', './base-client');
      }
    });
    
    if (content !== original) {
      fs.writeFileSync(filePath, content);
      console.log(`Updated ${filePath}`);
    }
  }
});
